common/flockfile/src/windows.rs

```diff
@@ -0,0 +1,32 @@
+use std::{
+    io,
+    path::{Path, PathBuf},
+};
+
+#[derive(thiserror::Error, Debug)]
+pub enum FlockfileError {
+    #[error(transparent)]
+    IoError(#[from] std::io::Error),
+}
+
+#[derive(Debug)]
+pub struct Flockfile {
+    pub path: PathBuf,
+}
+
+impl Flockfile {
+    pub fn new_lock(lock_name: impl AsRef<Path>) -> Result<Flockfile, FlockfileError> {
+        let path = Path::new("/no/lock/on/windows").join(lock_name);
+        Ok(Flockfile { path: path })
```

Comment:
Why not just directly return some error here instead of this dummy implementation? Or even `unimplemented!();` ? I thought we just wanted to make the code compile by just defining the APIs in Windows scope. We're not expecting this crate to be really executed by some other code on Windows, right?

Reply:
I expect this code to be executed in tests.